chcemvediet/apps/inforequests/views/attachment.py

```diff
@@ -49,3 +50,11 @@ def attachment_download(request, attachment_pk):
         raise Http404()
 
     return attachments_views.download(request, attachment)
+
+@require_http_methods([u'HEAD', u'GET'])
+def attachment_finalization_download(request, attachment_finalization_pk):
+    attachment_finalization = AttachmentFinalization.objects.get_or_404(
+            pk=attachment_finalization_pk)
+    if not attachment_finalization.attachment.generic_object.branch.inforequest.published:
+        raise Http404()
```

Comment:
Au. To si si isty, ze ti ten vyraz `attachment_finalization.attachment.generic_object.branch.inforequest.published` nikdy nepadne? Si si isty, ze vsetky properties cestou si required?

Dereferencovat objekt mozes iba ked si si isty jeho typom a ze ma danu property, inac ti to cele popada.

Reply:
@martinmacko47 Vsetky properties su `ForeignKey(null=False)`, atribut published je `Boolean(null=False, defalut=False)`. Attachment, ktore su anonymizovane maju generic_object (zatial) len `Action`. 
```
    attachment = (Attachment.objects
                  .attached_to(Action)
                  .not_normalized()
                  .first()
)
```
Do buducna, ako si hovoril, to mozu byt aj ine typy generic_object. Tak som pridal podmienku na jeho kontrolu.
```
    if isinstance(generic_object, Action):
        if generic_object.branch.inforequest.published:
            return attachments_views.download(request, attachment_finalization)
    raise Http404()
```

Reply:
@viliambalaz Ano, ak teda vsetky relacie po ceste su zarunene `NOT NULL`, tak potom staci overil len, ze `attachment_finalization.attachment.generic_object` je `Action`.